include/fixed_containers/fixed_graph.hpp

```diff
@@ -0,0 +1,1392 @@
+        for (NodeIndex i = 0; i < next_index_; ++i) {
+            comp.add_node(index_to_node_[i]);
+        }
+
+        for (NodeIndex u = 0; u < next_index_; ++u) {
+            for (NodeIndex v = 0; v < next_index_; ++v) {
+                if (u != v && !has_edge(u, v)) {
+                    if constexpr (std::is_void_v<EdgeType>) {
+                        comp.add_edge(u, v);
+                        if constexpr (!DIRECTED) comp.add_edge(v, u);
+                    } else {
+                        comp.add_edge(u, v, EdgeType{});
+                        if constexpr (!DIRECTED) comp.add_edge(v, u, EdgeType{});
+                    }
+                }
+            }
+        }
+
+        return comp;
+    }
+
+    // Graph union
+    constexpr FixedGraph graph_union(const FixedGraph& other) const
+    {
+        FixedGraph result = *this;
+
+        // Add nodes from other graph
+        for (NodeIndex i = 0; i < other.next_index_; ++i) {
+            result.add_node(other.index_to_node_[i]);
+        }
+
+        // Add edges from other graph
+        for (NodeIndex u = 0; u < other.next_index_; ++u) {
+            for (const auto& edge : other.neighbors(u)) {
+                NodeIndex v = other.get_neighbor(edge);
+                if constexpr (std::is_void_v<EdgeType>) {
+                    result.add_edge(u, v);
+                } else {
+                    result.add_edge(u, v, other.get_weight(edge));
+                }
+            }
+        }
+
+        return result;
+    }
+
+    // Graph intersection
+    constexpr FixedGraph graph_intersection(const FixedGraph& other) const
+    {
+        FixedGraph result{};
+
+        // Add common nodes
+        for (NodeIndex i = 0; i < next_index_; ++i) {
+            for (NodeIndex j = 0; j < other.next_index_; ++j) {
+                if (index_to_node_[i] == other.index_to_node_[j]) {
+                    result.add_node(index_to_node_[i]);
+                    break;
+                }
+            }
+        }
+
+        // Add common edges
+        for (NodeIndex u = 0; u < result.next_index_; ++u) {
+            for (NodeIndex v = 0; v < result.next_index_; ++v) {
+                if (has_edge(u, v) && other.has_edge(u, v)) {
+                    if constexpr (std::is_void_v<EdgeType>) {
+                        result.add_edge(u, v);
+                    } else {
+                        // Find the weight from this graph
+                        for (const auto& edge : neighbors(u)) {
+                            if (get_neighbor(edge) == v) {
+                                result.add_edge(u, v, get_weight(edge));
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        return result;
+    }
+
+    // Serialize graph to binary format
+    template <typename OutputIterator>
+    constexpr OutputIterator serialize(OutputIterator out) const
```

Comment:
Is there a strong justification for doing this (de)serialization by hand instead of using an existing format like flatbuffers or Avro? What is the use-case here?